backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,334 @@
+					},
+				},
+			}
+		}
+	}
+
+	if v, ok := data.GetOk("local_cert_ca_file"); ok {
+		// If local_cert_ca_file exists, the address must be of type HTTPS
+		if !isHTTPS(addressURL) {
+			return fmt.Errorf("Address must be of type HTTPS if local_cert_ca_file is set")
+		} //else {
+		if data.Get("mutual_tls_authentication").(bool) == true {
+			return fmt.Errorf("mutual_tls_authentication is true and local_cert_ca_file is set. Please choose one or the other")
+		}
+		if data.Get("skip_cert_verification").(bool) == true {
+			return fmt.Errorf("skip_cert_verification is true and local_cert_ca_file is set. Please choose one or the other")
+		}
+		b.localCertCAFile = v.(string)
+
+		// Get the SystemCertPool, continue with an empty pool on error
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		// Read in the ca cert file
+		cert, err := ioutil.ReadFile(b.localCertCAFile)
+		if err != nil {
+			return fmt.Errorf("Failed to read %s into memory: %s", b.localCertCAFile, err)
+		}
+		// Append our cert to the system pool
+		if ok := rootCAs.AppendCertsFromPEM(cert); !ok {
+			return fmt.Errorf("No certs could be appended: %s", cert)
+		}
+		// Replace the client with one that contains the CA.
+		client = &http.Client{
+			Timeout: time.Second * 10,
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: rootCAs,
+				},
+			},
+		}
+		//	}
+	}
+
+	if v, ok := data.GetOk("mutual_tls_authentication"); ok {
+		b.mutualTLS = v.(bool)
+
+		if b.mutualTLS {
+			// If mutual_tls_authentication = true, the address must be of type HTTPS
+			if !isHTTPS(addressURL) {
+				return fmt.Errorf("Address must be of type HTTPS if mutual_tls_authentication = true")
+			}
+			// If mutual_tls_authentication = true, the local_cert_file needs to be set.
+			if v, ok := data.GetOk("local_cert_file"); ok {
+				b.localCertFile = v.(string)
+			} else {
+				return fmt.Errorf("mutual_tls_authentication is true and local_cert_file is not set %s", b.localCertFile)
+			}
+			// If mutual_tls_authentication = true, the local_key_file needs to be set.
+			if v, ok := data.GetOk("local_key_file"); ok {
+				b.localKeyFile = v.(string)
+			} else {
+				return fmt.Errorf("mutual_tls_authentication is true and local_key_file is not set %s", b.localKeyFile)
+			}
+			// If mutual_tls_authentication = true, the local_ca_file needs to be set.
+			if v, ok := data.GetOk("local_ca_file"); ok {
+				b.localCAFile = v.(string)
+			} else {
+				return fmt.Errorf("mutual_tls_authentication is true and local_ca_file is not set %s", b.localCAFile)
+			}
+			// load client cert
+			certs, err := tls.LoadX509KeyPair(b.localCertFile, b.localKeyFile)
+			if err != nil {
+				return fmt.Errorf("Can not load pem files: %s and : %s. Error: %s", b.localCertFile, b.localKeyFile, err)
+			}
+			// Get the SystemCertPool, continue with an empty pool on error
+			rootCAs, _ := x509.SystemCertPool()
```

Comment:
again, why are you supressing/ignoring the error here?

Reply:
continue with an empty pool on error? We don't really care it was an error, we will create a brand new empty pool.
That's how I saw it was implemented by different people...
https://gist.github.com/olekukonko/6e864ba3ba84f19070a03627fe8052a5
https://forfuncsake.github.io/post/2017/08/trust-extra-ca-cert-in-go-app/

Reply:
No, I did not mean continue with an empty pool on error. I mean why are you checking that the `rootCAs` is `nil` and ignoring the error rather than checking if the error returned is not `nil`. 

After digging through the source code for that function call, it seems that if an error is returned `rootCAs` is always going to be `nil`. In some instances, when an error is returned the other return values are not guaranteed to be the zero value. Since this is not the case, it is sufficient to check that `rootCAs` is not `nil`.

However, if this behavior changes this check will not be sufficient. I guess its acceptable for now.